nengo/dists.py

```diff
@@ -350,11 +373,53 @@ class SubvectorLength(SqrtBeta):
     --------
     SqrtBeta
     """
+
     def __init__(self, dimensions, subdimensions=1):
         super(SubvectorLength, self).__init__(
             dimensions - subdimensions, subdimensions)
 
 
+class CosineSimilarity(SubvectorLength):
+    """Distribution of dot products between random unit vectors.
```

Comment:
Done. But "norm of the dot product" -> just "dot product" (note the result is signed, unlike `SubVectorLength`).